cli/popper/commands/cmd_run.py

```diff
@@ -97,6 +99,12 @@ def run_pipeline(project_root, pipeline, timeout, skip):
             for t in ['.err', '.out']:
                 with open('popper_logs/{}{}'.format(stage, t), 'r') as f:
                     pu.info(f.read())
+
+            if pu.get_filename(abs_path, 'teardown') and \
+               (not skip or 'teardown' not in skip.split(',')) and \
+               'teardown' not in is_executed:
+                execute(pu.get_filename(abs_path, 'teardown'), timeout)
+
```

Comment:
this could be simplified (`is_executed` list wouldn't be needed):

```python
if stage != 'teardown' and 'teardown' in pipeline['stages']:
    execute(pu.get_filename(abs_path, 'teardown'), timeout)
```

Reply:
Sorry, but I don't get how does it solve the problem

Reply:
do you want to keep track of what's has been executed with the goal of not running `teardown` when `teardown` fails? or is there another reason?

Reply:
I mean if by any chance `teardown` is ahead of a stage which fails, then `teardown` has already been run and after the stage fails we'll again run `teardown`. That is why I want to keep track of what has been executed.

Reply:
@ankan17 instead of keeping a whole list, maybe you could just use a flag variable to check if teardown has been executed or not. 

For example something like :- 
```
is_executed = 0 #outside the for-loop
if stage != 'teardown' and 'teardown' in pipeline['stages'] and is_executed == 0 :
    is_executed = 1
    execute(pu.get_filename(abs_path, 'teardown'), timeout)
```

Reply:
thanks @ankan17 and @UjjwalAyyangar . The idea is that `teardown`, if used, is supposed to be the last stage of a pipeline. While it is possible to have it before than the last one, that wouldn't be the way we intend users to make use of this functionality when defining pipelines.

So maybe what we could do is check on `popper init` if a user is defining a pipeline with `teardown` _not_ being the last stage. E.g.:

```bash
popper init --stages=one,teardown,three mypipe
ERROR: teardown should be the last stage. Consider renaming it or putting it at the end.
```

what do you guys think?

Reply:
@ivotron Yes yes, that makes a lot of sense. I will add in a check for this